File: DailyLeetcoding/150-evaluate-reverse-polish-notation/evaluate-reverse-polish-notation.java
class Solution {
    public int evalRPN(String[] tokens) {
        Stack<Integer> stack = new Stack<>();

        for(int i = 0; i < tokens.length; i++){
            if( tokens[i].equals("+") || tokens[i].equals("-") || tokens[i].equals("*") || tokens[i].equals("/") ){
                int val1 = stack.pop();
                int val2 = stack.pop();
                switch(tokens[i]){
                    case "+":
                        stack.push(val2 + val1);
                        break;
                    case "-":
                        stack.push(val2 - val1);
                        break;
                    case "*":
                        stack.push(val2 * val1);
                        break;
                    case "/":
                        stack.push(val2 / val1);
                        break;
                    default:
                }
            } else {
                stack.push(Integer.valueOf(tokens[i]));
            }
        }

        return stack.peek();
    }
}

File: DailyLeetcoding/150-evaluate-reverse-polish-notation/evaluate-reverse-polish-notation.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        stack = []

        for token in tokens:
            if token in {"+", "-", "/", "*"}:
                a = int(stack.pop())
                b = int(stack.pop())
                
                if token == "+":
                    result = b + a
                elif token == "*":
                    result = b * a
                elif token == "/":
                    result = int(b / a)
                elif token == "-":
                    result = b - a
                
                stack.append(result)
            else:
                stack.append(int(token))
        
        return stack[0]